.github/actions/with-docker/action.yml

```diff
@@ -16,21 +22,24 @@ runs:
     env:
       TAG_NAME: ${{ inputs.tag }}
       SUBDIR: ${{ inputs.subdir }}
+      BASE_IMAGE: ${{ inputs.distro }}
+      LLVM_VERSION: ${{ inputs.llvm }}
     run: |
-      set -x
+      set -euxo pipefail
 
       Z3_VERSION=4.8.15
       K_VERSION=$(cat ${SUBDIR}package/version)
 
-      docker build . --file ${SUBDIR}.github/workflows/Dockerfile.z3          --tag z3:${Z3_VERSION}
-      docker build . --file ${SUBDIR}.github/workflows/Dockerfile.stack-deps  --tag stack:${K_VERSION}
-      docker build . --file ${SUBDIR}.github/workflows/Dockerfile.maven-cache --tag maven:${K_VERSION}
+      docker build . --build-arg BASE_IMAGE=${BASE_IMAGE} --tag z3:${Z3_VERSION}   --file ${SUBDIR}.github/workflows/Dockerfile.z3
+      docker build . --build-arg BASE_IMAGE=${BASE_IMAGE} --tag stack:${K_VERSION} --file ${SUBDIR}.github/workflows/Dockerfile.stack-deps
+      docker build . --build-arg BASE_IMAGE=${BASE_IMAGE} --tag maven:${K_VERSION} --file ${SUBDIR}.github/workflows/Dockerfile.maven-cache
 
       docker build . --file ${SUBDIR}.github/workflows/Dockerfile    \
           --tag runtimeverification/${TAG_NAME}                      \
+          --build-arg BASE_IMAGE=${BASE_IMAGE}                       \
           --build-arg Z3_VERSION=${Z3_VERSION}                       \
           --build-arg K_VERSION=${K_VERSION}                         \
-          --build-arg LLVM_VERSION=14                                \
+          --build-arg LLVM_VERSION=${LLVM_VERSION}                   \
           --build-arg USER_ID=$(id -u) --build-arg GROUP_ID=$(id -g)
 
       docker run                             \
```

Comment:
The --user User:user  flag here will create all artifacts in the container as 'user' which doesn't exist within the host nodes shadow file. 

Checking some of the build agents (3/4) I did not see any files being created in the /opt/workspace/${SUBDIR} folder. This may not be an issue but I foresee we'll have upload/cleanup issues if anything eventually gets left behind by any workflows using the with-docker action.

Reply:
Well, not really. It specifically creates them as the same uid and gid as the lxc containers `actions-runner` user, not a sany specific user name.

The dockerfile itself sets up the `user:user` settings, so that it matches uid/gid of the external lxc container, that way the files that are created inside the container have premissions to be moved/deleted/modified outside the container.

Reply:
Gotcha, I see on 43 it's seeing those ID's as parameters to the build arguments